lib/src/rules/directives_ordering.dart

```diff
@@ -229,6 +302,31 @@ class _Visitor extends SimpleAstVisitor {
         .forEach(reportDirective);
   }
 
+  void _checkSectionInOrder(
+      Set<AstNode> lintedNodes, Iterable<NamespaceDirective> nodes) {
+    void reportDirective(NamespaceDirective directive) {
+      if (lintedNodes.add(directive)) {
+        rule._reportLintWithDirectiveSectionOrderedAlphabeticallyMessage(
+            directive);
+      }
+    }
+
+    List<NamespaceDirective> addIfUpperBound(
+        List<NamespaceDirective> acc, NamespaceDirective curr) {
+      if (acc.isEmpty || acc.last.uriContent.compareTo(curr.uriContent) < 0) {
+        acc.add(curr);
+      }
+
+      return acc;
+    }
+
+    final inOrderNodes = nodes.fold(<NamespaceDirective>[], addIfUpperBound);
+
+    nodes.toSet()
+      ..removeAll(inOrderNodes)
+      ..forEach(reportDirective);
```

Comment:
> I think that is a pretty good solution,other one could be calculate the nodes you have to move with minimal cost of sortering a block moving directives from its line to another then mark them, but it is more sophisticated and hard to implement, what do you think?

Yeah, I considered more complex algorithms (such as implementing `diff`), but I don't think the complexity of the implementation is warranted by any possible improvement in the UX.

I don't know for sure, but I'm still thinking it might be easier for users to be told once that the imports are not ordered well and then let them use a quick fix (or similar auto sorting technique) to fix the problem.

That said, I'm happy to let you continue as you're going and see what users actually think when they try using it. I could easily be wrong.